zipline/gens/tradesimulation.py

```diff
@@ -225,6 +225,20 @@ def _process_snapshot(self, dt, snapshot, instant_fill):
             elif event.type == DATASOURCE_TYPE.SPLIT:
                 self.algo.blotter.process_split(event)
 
+            elif event.type == DATASOURCE_TYPE.LIQUIDATION:
+                # cancel all open orders, if applicable
+                if event.sid in self.algo.blotter.open_orders:
+                    for order in self.algo.blotter.open_orders[event.sid]:
+                        self.algo.blotter.cancel(order.id)
+
+                # create new liquidation order, if applicable
+                if event.sid in self.algo.portfolio.positions:
+                    amt = self.algo.portfolio.positions[event.sid].amount
+                    self.algo.order(event.sid, -1 * amt)
```

Comment:
Doing this as an order gets the right values in the right places, but I worry that by using order there will be unexpected effects at some point, e.g. if this was used in a live context, this would attempt to order post the stocks liquidation event.

Seems like there needs to be a method on the portfolio such as `self.algo.portfolio.clear_out(sid)` or on the blotter class so that there can be hooks for there making sure that the position is cleaned out there.